test/unit-test/sigv4_utest.c

```diff
@@ -64,11 +193,191 @@ void formatAndVerifyInputDate( const char * pInputDate,
     tearDown();
 }
 
+/*==================== OpenSSL Based implementation of Crypto Interface ===================== */
+
+static int32_t valid_sha256_init( void * pHashContext )
+{
+    if( SHA256_Init( ( SHA256_CTX * ) pHashContext ) == 1 )
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+static int32_t valid_sha256_update( void * pHashContext,
+                                    const char * pInput,
```

Comment:
Heads up, I'm changing the type of this in the crypto interface from `const char *` to `const uint8_t *` in my PR, so you  would have to change this here too if that goes in first.